hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1042,9 +1048,17 @@ func (r *HostedClusterReconciler) reconcileControlPlaneOperator(ctx context.Cont
 	}
 
 	// Reconcile operator deployment
+	var pullSecret corev1.Secret
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: hcluster.Namespace, Name: hcluster.Spec.PullSecret.Name}, &pullSecret); err != nil {
+		return fmt.Errorf("failed to get pull secret: %w", err)
+	}
+	controlPlaneOperatorImage, err := getControlPlaneOperatorImage(hcluster, r.releaseProvider, r.HostedControlPlaneOperatorImage, pullSecret.Data[corev1.DockerConfigJsonKey])
```

Comment:
I guess maybe a byte slice is special cased enough places it won't matter (e.g. `len` checks on `nil` slice is ok), but it does seem to assume safe `nil` handling in all the subsequent code — maybe worth validating the key exists and throwing an error rather than passing junk data and getting an indirect error?

Reply:
good idea